src/net_processing.cpp

```diff
@@ -557,17 +557,20 @@ static void MaybeSetPeerAsAnnouncingHeaderAndIDs(NodeId nodeid, CConnman* connma
         }
         connman->ForNode(nodeid, [connman](CNode* pfrom){
             AssertLockHeld(cs_main);
-            uint64_t nCMPCTBLOCKVersion = (pfrom->GetLocalServices() & NODE_WITNESS) ? 2 : 1;
+            // Never try to upgrade a pre-segwit node to high-bandwidth compact block provider
```

Comment:
Here or in PR commit maybe lay out the rational "As a pre-segwit node won't send us new consensus data there is no reason to receive a block from them after softfork activation, for which we won't able to fully validate" (unless there is no segwit transaction in the block but did this happen these days?).

Reading more `GetFetchFlags`, did you spend time thinking why there is any reason we want to INV(MSG_BLOCK) no pre-segwit nodes ?

Reply:
> As a pre-segwit node won't send us new consensus data there is no reason to receive a block from them after softfork activation, for which we won't able to fully validate

I don't think this is the place to document such a fundamental fact about segwit - that we won't download blocks from pre-segwit nodes.

> Reading more GetFetchFlags, did you spend time thinking why there is any reason we want to INV(MSG_BLOCK) no pre-segwit nodes ?

I'm afraid I don't understand this question. It doesn't seem related to this PR, at least.